### PR TITLE
Story main

### PR DESCRIPTION
**Pull request by ajhswan**

### CODE REVIEW

File: functions.php [hunk 1]
```diff
@@ -0,0 +1,38 @@
+<?php
+
+/** dbQuery function to fetch data from database
+ * @param string $query
```
Comment (charliecog) [inline functions.php:5]: space between description and params, also between params and return
(please do for all docblocks)

File: functions.php [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+ * @param PDO $db
+ * @return array
+ */
+function query(string $query, PDO $db):array {
```
Comment (charliecog) [inline functions.php]: too generic - nothing wrong with giving a specific function a purpose eg: getReciepts()

File: index.php [hunk 3]
```diff
@@ -0,0 +1,33 @@
+<?php
+
+require_once 'dbConnect.php';
+require_once 'functions.php';
+
+$db = connectToDb($dbName, $userName, $password);
+
+//set query variable and pass to query function to fetch data from DB
+$sqlQuery = 'SELECT `id`, `supplier_name`, `date`, `amount`,`ccy`, `details`
```
Comment (charliecog) [inline index.php]: move this into inside your newly created getReceipts() function so you dont have to pass it in

File: index.php [hunk 4]
```diff
@@ -0,0 +1,33 @@
+<?php
+
+require_once 'dbConnect.php';
+require_once 'functions.php';
+
+$db = connectToDb($dbName, $userName, $password);
```
Comment (charliecog) [inline index.php]: $dbName, userName and password should all be defined inside the function - they have the appearence here of appearing out of nowhere so make it feel confusing

File: index.php [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+
+//set query variable and pass to query function to fetch data from DB
+$sqlQuery = 'SELECT `id`, `supplier_name`, `date`, `amount`,`ccy`, `details`
+FROM `receiptRecord`';
+$result = query($sqlQuery, $db);
+
+$receipt = displayData($result);
```
Comment (charliecog) [inline index.php]: should be plural $receipts

File: tests/unit.php [hunk 6]
```diff
@@ -0,0 +1,63 @@
+<?php
```
Comment (charliecog) [inline tests/unit.php]: this file should be called functions.php not unit.php

**Review by charliecog — COMMENTED** [review]
Looks good - almost there! last few bits to change

File: functions.php (hunk 2 above)
Comment (ajhswan) [reply functions.php]: done

File: tests/unit.php (hunk 6 above)
Comment (ajhswan) [reply tests/unit.php]: done

File: index.php (hunk 5 above)
Comment (ajhswan) [reply index.php]: done

File: index.php (hunk 4 above)
Comment (ajhswan) [reply index.php]: updated

**Review by ajhswan — COMMENTED** [review]
changes made - please can you check and let me know if it all good to pass review

File: tests/unit.php (hunk 6 above)
Comment (charliecog) [reply tests/unit.php]: its still called unit.php?
Have you pushed the changes?

File: functions.php (hunk 1 above)
Comment (charliecog) [reply functions.php:5]: ?

**Review by charliecog — APPROVED** [review]

**Pull request merged**